exercises/julia-1-a/runtests.jl

```diff
@@ -0,0 +1,214 @@
+using Test
+
+include("tictactoe.jl")
+
+@testset "Finished games where X won via colum victory" begin
+    boards = [
+        """
+        XOO
+        X  
+        X  
+        """,
+        """
+        OXO
+         X 
+         X 
+        """,
+        """
+        OOX
+          X
+          X
+        """
+    ]
+
+    for b in boards
+        @test gamestate(b) == :win
+    end
+end
+
+@testset "Finished games where O won via colum victory" begin
+    boards = [
+        """
+        OXX
+        OX 
+        O  
+        """,
+        """
+        XOX
+         OX
+         O 
+        """,
+        """
+        XXO
+         XO
+          O
+        """
+    ]
+
+    for b in boards
+        @test gamestate(b) == :win
+    end
+end
+
+@testset "Finished games where X won via row victory" begin
+    boards = [
+        """
+        XXX
+        XOO
+        O  
+        """,
+        """
+        O O
+        XXX
+         O 
+        """,
+        """
+         OO
+        O X
+        XXX
+        """
+    ]
+
+    for b in boards
+        @test gamestate(b) == :win
+    end
+end
+
+@testset "Finished games where O won via row victory" begin
+    boards = [
+        """
+        OOO
+        XXO
+        XX 
+        """,
+        """
+        XX 
+        OOO
+        X
+        """,
+        """
+        XOX
+         XX
+        OOO
+        """
+    ]
+
+    for b in boards
+        @test gamestate(b) == :win
+    end
+end
+
+@testset "Finished games where X won via diagonal victory" begin
+    boards = [
+        """
+        XOO
+         X 
+          X
+        """,
+        """
+        O X
+        OX 
+        X  
+        """
+    ]
+
+    for b in boards
+        @test gamestate(b) == :win
+    end
+end
+
+@testset "Finished games where O won via diagonal victory" begin
+    boards = [
+        """
+        OXX
+        OOX 
+        X O
+        """,
+        """
+          O
+         OX
+        OXX
+        """
+    ]
+
+    for b in boards
+        @test gamestate(b) == :win
+    end
+end
+
+@testset "Draws" begin
+    boards = [
+        """
+        XOX
+        XXO
+        OXO
+        """,
+        """
+        XXO
+        OOX
+        OXX
+        """
+    ]
+
+    for b in boards
+        @test gamestate(b) == :draw
+    end
+end
+
+@testset "Not-yet-finished games" begin
+    boards = [
+        """
+           
+         X  
+           
+        """,
+        """
+        O  
+         X
+           
+        """,
+        """
+        X  
+         XO
+        OX 
+        """
+    ]
+
+    for b in boards
+        @test gamestate(b) == :ongoing
+    end
+end
+
+@testset "Invalid boards" begin
+    boards = [
+        """
+        XX 
+           
+           
+        """,
+        """
+        OOX
```

Comment:
This is only an invalid state under the assumption that `X` starts which we've taken above. This is the only place where the test suite relies on that assumption to hold true, so I could also remove the test.